packages/synapse-react-client/src/components/error/SendMessageToEntityOwnerDialog.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useCallback, useState } from 'react'
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  TextField,
+  Typography,
+} from '@mui/material'
+import { DialogBaseTitle } from '../DialogBase'
+import {
+  useGetCurrentUserProfile,
+  useSendMessageToEntityOwner,
+} from '../../synapse-queries'
+import { displayToast } from '../ToastMessage'
+
+export const CONTACT_ADMIN_DIALOG_TITLE = 'Contact the Administrator'
+
+export type SendMessageToEntityOwnerDialogProps = {
+  isOpen: boolean
+  entityId: string
+  onHide: () => void
+}
+
+export default function SendMessageToEntityOwnerDialog(
+  props: SendMessageToEntityOwnerDialogProps,
+) {
+  const { onHide, entityId, isOpen } = props
+  const [message, setMessage] = useState<string>('')
+  const { data: currentUserProfile, isLoading: isLoadingCurrentUserProfile } =
+    useGetCurrentUserProfile()
+  const { mutate: sendMessage } = useSendMessageToEntityOwner({
+    onSuccess: () => {
+      onHide()
+      setMessage('')
+      displayToast('Message successfully sent')
+    },
+    onError: error => {
+      displayToast(`Message could not be sent: ${error.reason}`, 'danger')
+    },
+  })
+  const onSendMessage = useCallback(() => {
+    if (isLoadingCurrentUserProfile) {
+      console.error(
+        'Attempted to send message before user profile was loaded. This should never happen.',
+      )
+      return
+    }
+    if (!currentUserProfile) {
+      console.error(
+        'Attempted to send message without a user profile. This should never happen.',
+      )
+      return
+    }
+    const { firstName, lastName, userName } = currentUserProfile
+    sendMessage({
+      entityId,
+      subject: `${firstName} ${lastName} ${userName} Requested Access to ${entityId}`,
+      body: `Requesting access to https://www.synapse.org/Synapse:${entityId}\n\n${message}`,
+    })
+  }, [message, currentUserProfile, sendMessage])
+  return (
+    <Dialog maxWidth="md" open={isOpen} onClose={onHide} sx={{ zIndex: 1500 }}>
```

Comment:
Is the zIndex necessary?

Consider just using `ConfirmationDialog` for style consistency with other dialogs in SRC

Reply:
Removed zIndex, and converted to be a ConfirmationDialog!